ext/gd/gd.c

```diff
@@ -338,6 +338,13 @@ PHP_MINIT_FUNCTION(gd)
 	php_gd_object_minit_helper();
 	php_gd_font_minit_helper();
 
+#if defined(HAVE_GD_MEMORY_ALLOCATORS) && !ZEND_DEBUG
```

Comment:
What's the reason for now using `emalloc` with debug enabled? This can still be disabled by setting the `USE_ZEND_ALLOC=0` env variable for msan, valgrind and the like. This also doesn't match the condition in  `PHP_MSHUTDOWN_FUNCTION(gd)`.

Reply:
`_emalloc` needs more parameters when using `ZEND_DEBUG`, this is to make sure that we won't cause UB.

https://github.com/php/php-src/blob/02690fe3c0c35655bc40844780eba0c283a37a2f/Zend/zend_alloc.h#L69-L80

Also the `gdClearMemory*` methods of `gd` reset the allocation methods to the default ones, but this can be changed to match with what's in `PHP_MINIT_FUNCTION(gd)`.

Reply:
Of course, forgot about that, thanks 🙂 Wrapping them doesn't seem worth it.